profile/profile_test.go

```diff
@@ -689,6 +811,122 @@ func locationHash(s *Sample) string {
 	return tb
 }
 
+func TestInferUnits(t *testing.T) {
+	var tagFilterTests = []struct {
+		name             string
+		tagVals          []map[string][]int64
+		tagUnits         []map[string][]string
+		wantUnits        map[string]string
+		wantIgnoredUnits map[string][]string
+	}{
+		{
+			"One sample, multiple keys, different specified units",
+			[]map[string][]int64{{"key1": {131072}, "key2": {128}}},
+			[]map[string][]string{{"key1": {"bytes"}, "key2": {"kilobytes"}}},
+			map[string]string{"key1": "bytes", "key2": "kilobytes"},
+			map[string][]string{},
+		},
+		{
+			"One sample, one key with one value, unit specified",
+			[]map[string][]int64{{"key1": {8}}},
+			[]map[string][]string{{"key1": {"bytes"}}},
+			map[string]string{"key1": "bytes"},
+			map[string][]string{},
+		},
+		{
+			"bytes key, unit not specified",
```

Comment:
Nit: you start the test name with lowercase here, but with uppercase for the two other cases above, let's be consistent and use the uppercase here and in others.